lua/autorun/client/cl_devtools.lua

```diff
@@ -0,0 +1,133 @@
+local ranOnce = !ranOnce
+if (!ranOnce) then return end -- just to prevent edge cases
+ranOnce = true
+
+local isDev = isDev
+local refreshTime = 2
+local rootFolder = rootFolder or nil
+local fileTimes = fileTimes or {}
+local devReplStr = nil -- classname of all child files, taken from folder
+
+local function replaceFunction(base, regex, replacement)
+	local iStart, iEnd = string.find(base, regex, 1)
+
+	while iStart do
+		local xStart, _ = string.find(base, "(", iEnd, true) -- func name index
+		base = string.sub(base, 1, iStart-1) ..
+			replacement .. "." .. string.sub(base, iEnd + 1, xStart-1) ..
+			" = function" .. string.sub(base, xStart)
+		iStart, iEnd = string.find(base, regex, xStart)
+	end
+	return base
+end
+
+local function refreshFile(path)
+	if (!file.Exists(path, "LUA")) then return print("File " .. path .. " doesn't exist") end
+	if (!isDev) then return end
+	local code = file.Read(path, "LUA")
+
+	-- current filename as class format
+	local curClass = devReplStr or string.TrimRight(string.gsub(path, "^.*/", ""), ".lua")
+
+	-- simple replacement for compatibility, handling weapons and ents, no need for a lexer
+	-- performance impact should be minimal, it's ran once for testing, anyway
+	local storedStr = ".GetStored(\"" .. curClass .. "\")"
+
+	-- replace variables
+	code = string.gsub(string.gsub(code,
+		"%f[%a?]SWEP%.", "weapons" .. storedStr .. "."),
+		"%f[%a?]ENT%.", "scripted_ents" .. storedStr .. ".")
+
+	-- set all function definitions to assign to table
+	code = replaceFunction(replaceFunction(code,
```

Comment:
I think it would be easier to just set the global SWEP = scripted_weapons.GetStored(), then run the file. However, you do need to figure out how to handle `include` - either overwrite it globally or gsub it in the code string